src/dotnet/ExecutablePackageObtainer/ExecutablePackageObtainer.cs

```diff
@@ -0,0 +1,186 @@
+                        Directory.GetDirectories(
+                            individualToolVersion.WithSubDirectories(packageId).Value).Single()).Name;
+                DirectoryPath concreteVersionIndividualToolVersion =
+                    individualToolVersion.GetParentPath().WithSubDirectories(concreteVersion);
+                Directory.Move(individualToolVersion.Value, concreteVersionIndividualToolVersion.Value);
+
+                individualToolVersion = concreteVersionIndividualToolVersion;
+                packageVersion = concreteVersion;
+            }
+
+            ToolConfiguration toolConfiguration = GetConfiguration(packageId, packageVersion, individualToolVersion);
+
+            return new ToolConfigurationAndExecutableDirectory(
+                toolConfiguration,
+                individualToolVersion.WithSubDirectories(
+                    packageId,
+                    packageVersion,
+                    "tools",
+                    targetframework));
+        }
+
+        private static ToolConfiguration GetConfiguration(
+            string packageId,
+            string packageVersion,
+            DirectoryPath individualToolVersion)
+        {
+            FilePath toolConfigurationPath =
+                individualToolVersion
+                    .WithSubDirectories(packageId, packageVersion, "tools")
+                    .WithFile("DotnetToolsConfig.xml");
+
+            ToolConfiguration toolConfiguration =
+                ToolConfigurationDeserializer.Deserialize(toolConfigurationPath.Value);
+
+            return toolConfiguration;
+        }
+
+        private void InvokeRestore(
+            FilePath? nugetconfig,
+            FilePath tempProjectPath,
+            DirectoryPath individualToolVersion)
+        {
+            _projectRestorer.Restore(tempProjectPath, individualToolVersion, nugetconfig);
+        }
+
+        private FilePath CreateTempProject(
+            string packageId,
+            PackageVersion packageVersion,
+            string targetframework,
+            DirectoryPath individualToolVersion)
+        {
+            FilePath tempProjectPath = _getTempProjectPath();
+            if (Path.GetExtension(tempProjectPath.Value) != "csproj")
+            {
+                tempProjectPath = new FilePath(Path.ChangeExtension(tempProjectPath.Value, "csproj"));
+            }
+
+            EnsureDirectoryExists(tempProjectPath.GetDirectoryPath());
+            var tempProjectContent = new XDocument(
+                new XElement("Project",
+                    new XAttribute("Sdk", "Microsoft.NET.Sdk"),
+                    new XElement("PropertyGroup",
+                        new XElement("TargetFramework", targetframework),
+                        new XElement("RestorePackagesPath", individualToolVersion.Value),
+                        new XElement("DisableImplicitFrameworkReferences", "true")
+                    ),
+                    packageVersion.IsConcreteValue
```

Comment:
When the version is not concrete, is it possible to add a reference to version `*` and let nuget get the latest?  This would simplify things so we don't have to later do an add.

Reply:
This requires cooperation with nuget, I am using dotnet add package underneath today. But we may have a better interface later. So that may change. I'll keep that in mind